src/levanter/models/roberta.py

```diff
@@ -0,0 +1,852 @@
+        #     raise AttributeError("Model is being run as a MaskedLM aka an encoder model, but is_decoder is true")
+
+        k_rob, key_head = jrandom.split(key, 2)
+        roberta = RobertaModel.init(Vocab, config, add_pooling_layer=False, key=k_rob)
+        lm_head = RobertaLMHead.init(Vocab, config, key=key_head)
+
+        return RobertaForMaskedLM(roberta, lm_head, Vocab)
+
+    def get_output_embeddings(self):
+        return self.lm_head.decoder
+
+    def set_output_embeddings(self, new_embeddings):
+        self.lm_head.decoder = new_embeddings
+
+    def forward(
+        self,
+        input_ids: Optional[NamedArray] = None,
+        attention_mask: Optional[NamedArray] = None,
+        token_type_ids: Optional[NamedArray] = None,
+        position_ids: Optional[NamedArray] = None,
+        inputs_embeds: Optional[NamedArray] = None,
+        labels: Optional[NamedArray] = None,
+    ) -> Tuple[NamedArray]:
+        r"""
+        labels (`torch.LongTensor` of shape `(batch_size, sequence_length)`, *optional*):
+            Labels for computing the masked language modeling loss. Indices should be in `[-100, 0, ...,
+            config.vocab_size]` (see `input_ids` docstring) Tokens with indices set to `-100` are ignored (masked), the
+            loss is only computed for the tokens with labels in `[0, ..., config.vocab_size]`
+        kwargs (`Dict[str, any]`, optional, defaults to *{}*):
+            Used to hide legacy arguments that have been deprecated.
+        """
+
+        outputs = self.roberta(
+            input_ids,
+            attn_mask=attention_mask,
+            token_type_ids=token_type_ids,
+            position_ids=position_ids,
+            inputs_embeds=inputs_embeds
+        )
+
+        sequence_output = outputs[0]
+        prediction_scores = self.lm_head(sequence_output)
+
+        masked_lm_loss = None
+        if labels is not None:
+            masked_lm_loss = hnn.cross_entropy_loss(logits=prediction_scores, Label=self.Vocab, targets=labels)
+
+        output = (prediction_scores,)
+        return ((masked_lm_loss,) + output) if masked_lm_loss is not None else output
+    
+    
+def _rotate_half(x: NamedArray) -> NamedArray:
+    """Rotates half of the hidden dims of the input and concatenates them."""
+    HeadSize = x.axes[-1]
+    x1 = x[HeadSize, : HeadSize.size // 2]
+    x2 = x[HeadSize, HeadSize.size // 2 :]
+    out = hax.concatenate(HeadSize, (-x2, x1))
+    return out
+
+
+def _apply_rotary_pos_emb(
+    q: NamedArray,  # [batch, position, kv_heads, q_heads_per_group, head_size]
+    k: NamedArray,  # [batch, position, kv_heads, head_size]
+    cos: NamedArray,  # [position, head_size]
+    sin: NamedArray,  # [position, head_size]
+) -> Tuple[NamedArray, NamedArray]:
+    """Applies rotary position embedding to q and k."""
+    q_embed = q * cos + _rotate_half(q) * sin
+    k_embed = k * cos + _rotate_half(k) * sin
+    return q_embed, k_embed
+
+
+def llama_rotary_pos_emb(
```

Comment:
you'll want to delete this i think? (if you end up needing it, just import from llama.py)